platform/representation/src/MCTInclude.js

```diff
@@ -54,36 +54,42 @@ define(
          * @param {TemplateDefinition[]} templates an array of
          *        template extensions
          */
-        function MCTInclude(templates, $sce) {
+        function MCTInclude(templates, templateLinker) {
             var templateMap = {};
 
+            function link(scope, element) {
+                var changeTemplate = templateLinker.link(
+                    scope,
+                    element,
+                    scope.key && templateMap[scope.key]
+                );
+
+                scope.$watch('key', function (key) {
+                    changeTemplate(key && templateMap[key]);
+                });
+            }
+
             // Prepopulate templateMap for easy look up by key
             templates.forEach(function (template) {
                 var key = template.key,
-                    path = $sce.trustAsResourceUrl([
+                    path = [
```

Comment:
Hrm, do you think getPath could be provided by the TemplateLinker?  Same code as the representation uses below.  ~~Can also standardize the optimizations: `mct-include` precalculates all paths, while `mct-representation` calculates paths on demand (and repeatedly); an in between solution could be memoizing calls to getPath using a WeakMap.~~

On second though; I think performance optimizations here are unnecessary.